From 75485746a5050a535bf8995f4a0090fe9e8a9d6c Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Wed, 8 Feb 2017 09:37:25 +0100
Subject: [PATCH 1272/1566] xilinx: zynqmp: Panic if incorrect pmufw version
 is used

commit  cbd1025ec422e15fc8bb5a9a127aa6f2bd0f65e5 from
https://github.com/Xilinx/linux-xlnx.git

Also panic if zynqmp clock node is not present to ensure
that everybody is using correct and latest pmufw version and
also using zynqmp clock driver.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/xilinx/zynqmp/pm.c |   12 +++---------
 1 files changed, 3 insertions(+), 9 deletions(-)

diff --git a/drivers/soc/xilinx/zynqmp/pm.c b/drivers/soc/xilinx/zynqmp/pm.c
index 48a77b6..e912e4c 100644
--- a/drivers/soc/xilinx/zynqmp/pm.c
+++ b/drivers/soc/xilinx/zynqmp/pm.c
@@ -1221,29 +1221,23 @@ static int __init zynqmp_plat_init(void)
 	int ret = 0;
 
 	np = of_find_compatible_node(NULL, NULL, "xlnx,zynqmp-pm");
-	if (!np) {
-		pr_err("%s: pm node not found\n", __func__);
-		ret = -ENXIO;
-		goto np_err;
-	}
+	if (!np)
+		panic("%s: pm node not found\n", __func__);
 
 	get_set_conduit_method(np);
 
 	/* Check PM API version number */
 	zynqmp_pm_get_api_version(&pm_api_version);
 	if (pm_api_version != ZYNQMP_PM_VERSION) {
-		pr_err("%s power management API version error. Expected: v%d.%d - Found: v%d.%d\n",
+		panic("%s power management API version error. Expected: v%d.%d - Found: v%d.%d\n",
 		       __func__,
 		       ZYNQMP_PM_VERSION_MAJOR, ZYNQMP_PM_VERSION_MINOR,
 		       pm_api_version >> 16, pm_api_version & 0xffff);
-
-		do_fw_call = do_fw_call_fail;
 	}
 
 	pr_info("%s Power management API v%d.%d\n", __func__,
 		ZYNQMP_PM_VERSION_MAJOR, ZYNQMP_PM_VERSION_MINOR);
 
-np_err:
 	of_node_put(np);
 	return ret;
 }
-- 
1.7.5.4

